tests/queries/0_stateless/03591_optimize_prewhere_row_policy.sql

```diff
@@ -0,0 +1,31 @@
+-- {echoOn}
+
+SET use_query_condition_cache=0;
+
+DROP TABLE IF EXISTS 03591_test;
+
+DROP ROW POLICY IF EXISTS 03591_rp ON 03591_test;
+
+CREATE TABLE 03591_test (a Int32, b Int32) ENGINE=MergeTree ORDER BY tuple();
+
+INSERT INTO 03591_test VALUES (3, 1), (2, 2), (3, 2);
+
+SELECT * FROM 03591_test;
+
+SELECT * FROM 03591_test WHERE throwIf(b=1, 'Should throw') SETTINGS optimize_move_to_prewhere = 1; -- {serverError FUNCTION_THROW_IF_VALUE_IS_NON_ZERO}
+
+CREATE ROW POLICY 03591_rp ON 03591_test USING b=2 TO CURRENT_USER;
```

Comment:
It may be  good idea to have a bit more complex row policies, e.g. another PERMISSIVE and another RESTRICTIVE.